crates/cubecl-core/src/frontend/element/base.rs

```diff
@@ -382,5 +382,7 @@ pub(crate) fn __expand_new<C: Numeric, Out: Numeric>(
     val: C,
 ) -> ExpandElementTyped<Out> {
     let input: ExpandElementTyped<C> = val.into();
-    <Out as super::Cast>::__expand_cast_from(scope, input)
+    let const_val = input.expand.as_const().unwrap();
+    let var = Variable::constant(const_val.cast_to(Out::as_elem(scope)));
+    ExpandElement::Plain(var).into()
```

Comment:
Nice fix! Forcing the constant is the right move here.

Reply:
It was causing issues because I implemented the `IntExpand` dynamic element, and that was causing this code to emit `i64` constants in WGPU (which isn't supported).